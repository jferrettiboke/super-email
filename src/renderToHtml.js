import { renderToStaticMarkup } from "react-dom/server";
import inlineStyles from "./utils/inlineStyles";

const renderToHtml = element =>
  inlineStyles(`
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
${renderToStaticMarkup(element)}
`);

export default renderToHtml;
