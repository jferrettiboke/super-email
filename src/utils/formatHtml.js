import pretty from "pretty";

const formatHtml = html => pretty(html, { ocd: true });

export default formatHtml;
