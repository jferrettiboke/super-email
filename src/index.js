import renderToHtml from "./renderToHtml";
import Email from "./components/Email";
import Container from "./components/Container";
import Row from "./components/Row";
import Column from "./components/Column";
import Box from "./components/Box";
import Text from "./components/Text";
import Image from "./components/Image";
import Link from "./components/Link";

export {
  // Basic
  renderToHtml,
  Email,
  // Grid
  Container,
  Row,
  Column,
  // Other
  Box,
  Text,
  Image,
  Link
};
