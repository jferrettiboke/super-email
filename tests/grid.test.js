import React from "react";

import { renderToHtml, Email, Container, Row, Column } from "../src";
import formatHtml from "../src/utils/formatHtml";

test("renders successfully", () => {
  const html = renderToHtml(
    <Email title="Welcome">
      <Container>
        <Row>
          <Column width="6">Column 1</Column>
          <Column width="6">Column 2</Column>
        </Row>
      </Container>
    </Email>
  );
  expect(formatHtml(html)).toMatchSnapshot();
});
