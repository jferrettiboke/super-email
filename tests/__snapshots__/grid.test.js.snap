// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\" style=\\"margin: 0; padding: 0; border: none; box-sizing: border-box;\\">
  <head>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\">
    <title>Welcome</title>
  </head>
  <body style=\\"border: none; margin: 0; padding: 0; box-sizing: border-box; background-color: white;\\">
    <table data-component=\\"body\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 100%; height: 100%; background-color: white;\\"
      width=\\"100%\\" height=\\"100%\\" valign=\\"top\\" bgcolor=\\"white\\">
      <tr style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 100%;\\" valign=\\"top\\">
        <td align=\\"center\\" valign=\\"top\\" style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 100%; word-wrap: break-word; font-size: 16px; line-height: 1.3; color: #0a0a0a; font-weight: normal;\\"
          width=\\"100%\\">
          <center style=\\"margin: 0; padding: 0; border: none; box-sizing: border-box; width: 100%;\\">
            <table data-component=\\"container\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"box-sizing: border-box; margin: 0 auto; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 100%; max-width: 580px;\\"
              width=\\"100%\\" valign=\\"top\\">
              <tr style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 100%;\\" valign=\\"top\\">
                <td style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 100%;\\" width=\\"100%\\" valign=\\"top\\">
                  <table data-component=\\"row\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 100%; display: table;\\"
                    width=\\"100%\\" valign=\\"top\\">
                    <tr style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 100%;\\" valign=\\"top\\">
                      <td data-component=\\"column\\" style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 50%; margin-top: 0; margin-bottom: 0; margin-left: auto; margin-right: auto; padding-top: 0; padding-bottom: 0; padding-left: 15px; padding-right: 15px;\\"
                        width=\\"50%\\" valign=\\"top\\">Column 1</td>
                      <td data-component=\\"column\\" style=\\"box-sizing: border-box; margin: 0; padding: 0; border: none; border-spacing: 0; border-collapse: collapse; vertical-align: top; width: 50%; margin-top: 0; margin-bottom: 0; margin-left: auto; margin-right: auto; padding-top: 0; padding-bottom: 0; padding-left: 15px; padding-right: 15px;\\"
                        width=\\"50%\\" valign=\\"top\\">Column 2</td>
                    </tr>
                  </table>
                </td>
              </tr>
            </table>
          </center>
        </td>
      </tr>
    </table>
  </body>
</html>"
`;
